tests/common/plugins/sanity_check/recover.py

```diff
@@ -103,13 +105,27 @@ def _neighbor_vm_recover_bgpd(node=None, results=None):
     results[nbr_host.hostname] = result
 
 
-def neighbor_vm_restore(duthost, nbrhosts, tbinfo):
+def _neighbor_vm_recover_config(node=None, results=None):
+    if isinstance(node["host"], SonicHost):
+        config_reload(node["host"])
+    return results
+
+
+def neighbor_vm_restore(duthost, nbrhosts, tbinfo, result):
     logger.info("Restoring neighbor VMs for {}".format(duthost))
     mg_facts = duthost.get_extended_minigraph_facts(tbinfo)
     vm_neighbors = mg_facts['minigraph_neighbors']
     if vm_neighbors:
-        results = parallel_run(_neighbor_vm_recover_bgpd, (), {}, nbrhosts.values(), timeout=300)
-        logger.debug('Results of restoring neighbor VMs: {}'.format(json.dumps(dict(results))))
+        if result["check_item"] == "bgp":
```

Comment:
Would it be better to not affect original logic?
by:

if result["check_item"] == "neighbor_macsec_empty":
            unhealthy_nbrs = []
            for name, host in nbrhosts.items():
                if name in result["unhealthy_nbrs"]:
                    unhealthy_nbrs.append(host)
            parallel_run(_neighbor_vm_recover_config, (), {}, unhealthy_nbrs, timeout=300)
            logger.debug('Results of restoring neighbor VMs: {}'.format(unhealthy_nbrs))
else: 
            results = parallel_run(_neighbor_vm_recover_bgpd, (), {}, nbrhosts.values(), timeout=300)
            logger.debug('Results of restoring neighbor VMs: {}'.format(json.dumps(dict(results))))